Clinical-Genomics/scout
# Re-upload case variants with the option to keep custom actions (dismissed, manual rank, mosaic, acmg, comments)

This PR adds the functionality to keep the custom manual tags assigned by a user when the variants of a case are re-uploaded (fix #1649)

Custom actions that are kept upon variant re-upload are:
- manual_rank
- dismiss_variant
- mosaic_tags
- cancer_tier
- acmg_classification
- comments

**How to test**:
1. On a local instance of Scout choose one or more variants
1. Assign the following manual actions to the variant(s):
- [x] manual rank (even in two separate occasions, but remember the last rank assigned)
- [x] dismiss tag (one or more values, even in two separate occasions, but remember the assigned values)
- [x] mosaic tag (one or more values, even in two separate occasions, but remember the assigned values)
- [x] cancer_tier
- [x] acmg classification (manual)
- [x] acmg classification (via scout app)
1. From master branch use the following CLI command to re-upload the SNVs
`scout --demo load variants internal_id -f --snv`
and you'll see that the manual actions disappear
1. Switch to this branch and run the same command :
`scout --demo load variants internal_id -f --snv`
You should see that the actions from the old variants are propagated into the new ones.

Testing the propagation of comments requires reupload of a case on scout stage:
1. install the branch on hasta stage
1. Add a local and a global comment for a variant of a case on scout stage (clinical-db stage)
1. reupload the variants of a stage case
`scout --demo load variants internal_id -f --snv`
1. Make sure that the comments are present in the newly loaded variant
- [x] global comments if variant ID changes (Automatic tests)
- [x] local comments if variant ID changes (Automatic tests)


**Expected outcome**:
The functionality should be working

**Review:**
- [x] tests executed by  DNil, CR
